src/main/scala/ch/openolitor/stammdaten/repositories/StammdatenReadRepositoryAsync.scala

```diff
@@ -323,23 +323,38 @@ class StammdatenReadRepositoryAsyncImpl extends BaseReadRepositoryAsync with Sta
     getProduzentDetailReportQuery(id, projekt).future
   }
 
-  def getDepotlieferungAbos(implicit asyncCpContext: MultipleAsyncConnectionPoolContext, filter: Option[FilterExpr]): Future[List[DepotlieferungAbo]] = {
-    getDepotlieferungAbosQuery(filter).future
+  def getDepotlieferungAbos(xFlags: Option[AbosComplexFlags])(implicit asyncCpContext: MultipleAsyncConnectionPoolContext, filter: Option[FilterExpr]): Future[List[DepotlieferungAbo]] = {
+    xFlags match {
+      case Some(acf) if acf.zusatzAbosAktiv =>
+        getDepotlieferungAbosOnlyAktiveZusatzabosQuery(filter).future
```

Comment:
As discusses I would vote to merge them together and query alwaey zusatzabos by a join